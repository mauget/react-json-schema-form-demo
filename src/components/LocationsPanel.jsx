import React, { useRef } from 'react';
import styled from 'styled-components';
import Form from '@rjsf/core';
import toastr from 'toastr/toastr';
import jsonSchema from '../schemas/jsonSchema';
import uiSchema from '../schemas/uiSchema';
import getFormData from '../services/getFormData';


const StyledWrapper = styled.section`
    margin: 2rem;
    padding: 4rem;
    border: solid 1px darkred;
    color: darkslateblue;
    border-radius: 12px;
`;

const StyledTitle = styled.h1`
    font-size: 1.5rem;
    text-align: center;
`;

const StyledDetailArea = styled.div`
    margin: 1.5rem;
`;

export default function LocationsPanel() {
    const ref = useRef(getFormData());

    const submitHandler = () => {
        // eslint-disable-next-line no-console
        console.log(ref.current);
        toastr.success(JSON.stringify(ref.current));
    };

    const handleOnChange = (event) => {
        ref.current = event.formData;
    };

    return (
        <StyledWrapper>
            <StyledTitle>Demonstration of React JSON Schema</StyledTitle>
            <StyledDetailArea>
                <Form
                    dataTestId="testid"
                    schema={jsonSchema}
                    uiSchema={uiSchema}
                    formData={ref.current}
                    onChange={handleOnChange}
                    onSubmit={submitHandler}
                />
            </StyledDetailArea>
        </StyledWrapper>
    );
}
