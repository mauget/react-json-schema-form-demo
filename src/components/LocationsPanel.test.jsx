import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import LocationsPanel from './LocationsPanel';

configure({ adapter: new Adapter() });

describe('LocationsPanel', () => {

    it('should match its snapshot', () => {
        const wrapper = shallow(<LocationsPanel />);
        expect(wrapper).toMatchSnapshot();
    });

    it.skip('should call onChage', () => {
        // const onChange = jest.fn()
        const event = { target: { value: 'test' } };
        const wrapper = mount(<LocationsPanel />);
        wrapper.find('TextField').simulate('onChange', event)
    });

});
