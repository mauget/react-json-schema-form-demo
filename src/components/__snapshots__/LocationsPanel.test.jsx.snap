// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationsPanel should call onChage 1`] = `ShallowWrapper {}`;

exports[`LocationsPanel should match its snapshot 1`] = `ShallowWrapper {}`;
